src/check_datapackage/extensions.py

```diff
@@ -162,6 +163,7 @@ class RequiredCheck(BaseModel, frozen=True):
     jsonpath: JsonPath
     message: str
     _targets: list[TargetJsonPath] = PrivateAttr()
+    instance: Any = None
```

Comment:
@martonvago @lwjohnst86 I set the default to `None` so that we don't have to pass `error.instance` around everywhere that we create an Issue (since it is only used for the `explain` message right now), but if you think that approach is better, I can change to that.

Reply:
Hmm, thinking about this more, if `instance` is part of the `explain` message, we should actually require that it be present, no? Rather than default to `None`?

Reply:
Ok, tomorrow I can look into the other places where it is used and see if it makes sense to always include a message there. For example for the required and [custom check classes](https://github.com/seedcase-project/check-datapackage/blob/9410ee4881207ae124d86aeacefe4a4e4154a550/src/check_datapackage/extensions.py#L92-L94) (maybe you know already if that makes sense?).

Reply:
Needs a docstring for this attribute.